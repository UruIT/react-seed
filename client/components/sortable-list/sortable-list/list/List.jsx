import React from 'react';
import { SortableContainer } from 'react-sortable-hoc';
import Card from '../../../card';
import ListItem from '../list-item';

const List = SortableContainer(({ items }) => {
	return (
		<ul>
			{items.map((item, index) => (
				<ListItem key={`item-${index}`} index={index}>
					<Card title={item.title} content={item.content} />
				</ListItem>
			))}
		</ul>
	);
});

export default List;
