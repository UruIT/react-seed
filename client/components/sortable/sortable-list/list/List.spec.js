import React from 'react';
import renderer from 'react-test-renderer';

import List from './List';

const items = [
	{
		title: 'Item 1',
		content: 'Item number one'
	},
	{
		title: 'Item 2',
		content: 'Item number two'
	}
];

describe('<List />', () => {
	it('snapshot', () => {
		const tree = renderer.create(<List items={items} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
