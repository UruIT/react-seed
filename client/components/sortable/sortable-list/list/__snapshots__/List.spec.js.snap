// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> snapshot 1`] = `
<ul>
  <div>
    <div
      className="card"
    >
      <div
        className="cardInfo"
      >
        <h4>
          Item 1
        </h4>
        <p>
          Item number one
        </p>
      </div>
    </div>
  </div>
  <div>
    <div
      className="card"
    >
      <div
        className="cardInfo"
      >
        <h4>
          Item 2
        </h4>
        <p>
          Item number two
        </p>
      </div>
    </div>
  </div>
  <div>
    <div
      className="card"
    >
      <div
        className="cardInfo"
      >
        <h4>
          Item 3
        </h4>
        <p>
          Item number three
        </p>
      </div>
    </div>
  </div>
  <div>
    <div
      className="card"
    >
      <div
        className="cardInfo"
      >
        <h4>
          Item 4
        </h4>
        <p>
          Item number four
        </p>
      </div>
    </div>
  </div>
  <div>
    <div
      className="card"
    >
      <div
        className="cardInfo"
      >
        <h4>
          Item 5
        </h4>
        <p>
          Item number five
        </p>
      </div>
    </div>
  </div>
  <div>
    <div
      className="card"
    >
      <div
        className="cardInfo"
      >
        <h4>
          Item 6
        </h4>
        <p>
          Item number six
        </p>
      </div>
    </div>
  </div>
</ul>
`;
